src/data/nat/fib.lean

```diff
@@ -159,4 +160,15 @@ two_step_induction rfl rfl (λ n h1 h2, by
 { rw [fib_add_two, h1, h2, finset.nat.antidiagonal_succ_succ', finset.nat.antidiagonal_succ'],
   simp [choose_succ_succ, finset.sum_add_distrib, add_left_comm] })
 
+lemma fib_succ_eq_succ_sum (n : ℕ):
+  fib (n + 1) = (∑ k in finset.range n, fib k) + 1 :=
+begin
+  induction n with n ih,
+  { simp },
+  { calc fib (n + 2) = fib n + fib (n + 1) : by rw [fib_add_two]
+                 ... = fib n + (∑ k in finset.range n, fib k) + 1 : by rw ih; ring
+                 ... = (∑ k in finset.range (n + 1), fib k) + 1
+                          : by simp [finset.range_add_one] }
```

Comment:
You don't need `ring` here, `add_assoc` is enough:
```suggestion
  { calc fib (n + 2) = fib n + fib (n + 1)                        : fib_add_two
                 ... = fib n + (∑ k in finset.range n, fib k) + 1 : by rw [ih, add_assoc]
                 ... = (∑ k in finset.range (n + 1), fib k) + 1   : by simp [finset.range_add_one] }
```
similarly, you don't need to use `rw` for the first one, `fib_add_two` is precisely the statement of it.